### PR TITLE
Jorwoods/type hint data alert

### PR DESCRIPTION
**Pull request by jorwoods**
Add type hinting to data alert item and endpoint

### CODE REVIEW

File: tableauserverclient/models/data_alert_item.py [hunk 1]
```diff
@@ -9,44 +9,50 @@
...
+
+
 class DataAlertItem(object):
     class Frequency:
-        Once = "Once"
-        Frequently = "Frequently"
-        Hourly = "Hourly"
-        Daily = "Daily"
-        Weekly = "Weekly"
+        Once: str = "Once"
```
Comment (t8y8) [inline tableauserverclient/models/data_alert_item.py:20]: I think these enum-like classes don't really need the type hint, it's pretty obvious it's a string and isn't going to change. Mypy is smart enough to infer from the literal.

File: tableauserverclient/server/request_factory.py [hunk 2]
```diff
@@ -87,22 +92,26 @@ def update_req(self, column_item):
 
 
 class DataAlertRequest(object):
-    def add_user_to_alert(self, alert_item, user_id):
+    def add_user_to_alert(self, alert_item: "DataAlertItem", user_id: str) -> bytes:
```
Comment (t8y8) [inline tableauserverclient/server/request_factory.py:95]: This should be a string, not bytes

File: tableauserverclient/server/request_factory.py [hunk 3]
```diff
@@ -87,22 +92,26 @@ def update_req(self, column_item):
...
         xml_request = ET.Element("tsRequest")
         user_element = ET.SubElement(xml_request, "user")
         user_element.attrib["id"] = user_id
 
         return ET.tostring(xml_request)
 
-    def update_req(self, alert_item):
+    def update_req(self, alert_item: "DataAlertItem") -> bytes:
```
Comment (t8y8) [inline tableauserverclient/server/request_factory.py:102]: string, not bytes

**Review by t8y8 — CHANGES_REQUESTED** [review]
minor changes -- though as a general rule i think we can leave hints off the "enum-like" classes.

Now that we're on 3.5 and later, we could just convert them all to enums as well...

File: tableauserverclient/models/data_alert_item.py (hunk 1 above)
Comment (jorwoods) [reply tableauserverclient/models/data_alert_item.py]: Removed type hints from the frequency namespace

**Review by t8y8 — APPROVED** [review]

**Pull request merged**